### ISSUE TITLE
[BUG] Tooltip UI gets cut off for large recipes

### ISSUE THREAD

**Issue by lukaskollmer** [ShadowTheAge/yafc#208]
(I'm using the latest version of YAFC)

Not sure if this is actually a bug and not just a result of my screen potentially being too small, but large recipes (large w.r.t. their number of inputs and outputs) cause the tooltip view to be only partially visible, with the topmost (and bottommost?) parts being off-screen.

My screen size is 3024 × 1964 (14" MacBook Pro), which i guess shouldn't be too uncommon.
Idk how difficult this would be to change (if it is possible at all), so I don't wanna make unreasonable suggestions, but here's some ideas:
- IMO a simple fix for this would be to try to detect situations where the tooltip would be taller than the screen size, and then scale it down so that it fits.
- If the tooltip were made scrollable, you could allow scrolling while hovering over some tooltip-displaying item and then redirect these scroll events to the tooltip. Though I don't think that would work too well since you can currently already scroll in the production sheet view while hovering over a tooltip (and it'll just scroll the production sheet, which is probably more expected than it scrolling the tooltip).
- Instead of scaling the tooltips, they could be multi-column if the screen height isn't sufficient to display the whole thing.

![](https://lukaskollmer.s3.amazonaws.com/factorio/2023-02-16_yafc_tooltip_size.png)

**Comment by DaleStan** [comment]
I agree we shouldn't assume everyone has a 4k screen, but I'm not sure what you're doing here.

With a slightly different set of mods, I see this on my 2560×1440 screen:
![image](https://user-images.githubusercontent.com/21223975/219834886-1d3ae30a-ada2-446a-adab-406b16de16b4.png)

It would still have troubles on a 1920×1080 screen, but your 3024×1964 should be just fine.

**Comment by lukaskollmer** [comment]
i thinks it's because, while the native screen resolution is 3024×1964, it's rendered at 2x, so the effective resolution is actually just 1512x982

**Comment by veger** [comment]
I have similar issues on my laptop which has a smaller screen than my desktop monitors.

I have been thinking to take a stab at this, but since UI/SDL is not my strong suit I didn't think too much on the problem and potential solutions. Bit I think, as you also propose, a redesign of the tooltip is our best bet to fix this issue.

For my [summary view](https://github.com/ShadowTheAge/yafc/pull/145) I had to jump to hoops to figure out ways to find the height of the 'parent space' in order to make the 2D scrollable object fill this whole space. So I doubt scaling to fit the available space is going to be possible (again SDL is not my string suit, so I might have missed something obvious)
